File: back/src/main/java/com/projet/valatMoslehGrodet/dto/AccountDTO.java
package com.projet.valatMoslehGrodet.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccountDTO {
    private Long id;
    @NotNull
    private String firstname;
    @NotNull
    private String lastName;
    @NotNull
    private String password;
    @NotNull
    private String email;
    @NotNull
    private AddressDTO address;
    @NotNull
    private Date birthDate;
    private List<String> preferences;
    private Float rating;
    private List<CommentDTO> comments;
    private String description;
    private List<EventDTO> events;
}

File: front/src/lib/api/models/AccountDto.ts
import { AddressDto } from "./AddressDto";
import { CommentDto } from "./CommentDto";
import { EventDto } from "./EventDto";

export type AccountDto = {
  /**
   * @type integer | undefined, int64
   */
  id?: number;
  /**
   * @type string
   */
  firstname: string;
  /**
   * @type string
   */
  lastName: string;
  /**
   * @type string
   */
  password: string;
  /**
   * @type string
   */
  email: string;
  address: AddressDto;
  /**
   * @type string, date-time
   */
  birthDate: string;
  /**
   * @type array | undefined
   */
  preferences?: string[];
  /**
   * @type number | undefined, float
   */
  rating?: number;
  /**
   * @type array | undefined
   */
  comments?: CommentDto[];
  /**
   * @type string | undefined
   */
  description?: string;
  /**
   * @type array | undefined
   */
  events?: EventDto[];
};
